samples/apps/copilot-chat-app/webapp/tests/chat.test.ts

```diff
@@ -1,51 +1,53 @@
-loaded, and signed in. The user can send a message to the bot, and expect
-a response.
-*/
-test('get response from bot', async ({ page }) => {
-    // Make sure the server is running.
-    await page.goto('https://localhost:40443/healthz');
-    expect(page.getByText('Healthy')).toBeDefined();
+test.describe('Copilot Chat App Test Suite', () => {
+    // Note: A new chat session is opened for each test so that 
+    // the chat history is not polluted and the LLM is not confused.
 
-    await page.goto('/');
-    // Expect the page to contain a "Login" button.
-    await page.getByRole('button').click();
-    // Clicking the login button should redirect to the login page.
-    await expect(page).toHaveURL(new RegExp('^' + process.env.REACT_APP_AAD_AUTHORITY));
-    // Login with the test user.
-    await page.getByPlaceholder('Email, phone, or Skype').click();
-    await page.getByPlaceholder('Email, phone, or Skype').fill(process.env.REACT_APP_TEST_USER_ACCOUNT as string);
-    await page.getByRole('button', { name: 'Next' }).click();
-    await page.getByPlaceholder('Password').click();
-    await page.getByPlaceholder('Password').fill(process.env.REACT_APP_TEST_USER_PASSWORD as string);
-    await page.getByRole('button', { name: 'Sign in' }).click();
+    test.describe('A, runs in parallel with B', () => {
+        test.describe.configure({ mode: 'parallel' });
+        
+        // Server Tests
+        test('Server Health', async ({ page }) => { 
+            await simpletests.serverHealth(page) });
 
-    // Select No if asked to stay signed in.
-    const isAskingStaySignedIn = await page.$$("text='Stay signed in?'");
-    if (isAskingStaySignedIn) {
-        await page.getByRole('button', { name: 'No' }).click();
-    }
+        // Basic Operations
+        test('Basic Bot Responses', async ({ page }) => { 
+            test.setTimeout(util.TestTimeout);
+            await simpletests.basicBotResponses(page) });
+        test('Chat Title Change', async ({ page }) => { 
+            test.setTimeout(util.TestTimeout);
+            await simpletests.chatTitleChange(page) });
+        test('Chat Document Upload', async ({ page }) => { 
+            test.setTimeout(util.TestTimeout);
+            await simpletests.documentUpload(page) });
+    });
+        
+    test.describe('B, runs in parallel with A', () => {
+        test.describe.configure({ mode: 'parallel' });
+        
+        // Planner Testing
+        test('Planner Test: Klarna', async ({ page }) => { 
+            test.setTimeout(util.TestTimeout);
+            await plannertests.klarnaTest(page) });
+        test('Planner Test: Jira', async ({ page }) => { 
+            test.setTimeout(util.TestTimeout);
+            await plannertests.jiraTest(page) });
 
-    // After login, the page should redirect back to the app.
-    await expect(page).toHaveTitle('Copilot Chat');
+        // Todo: Action Planner intermittently returns a 400 error
```

Comment:
do we have already have a bug tracking this?

Reply:
We should. @teresaqhoang should be on it. I'll link it here when I find it.

Reply:
https://github.com/microsoft/semantic-kernel/issues/1726